### PR TITLE
Process manager is changed to PM2 instead of Foreman

### PR DESCRIPTION
**Pull request by changev**
use PM2 instead of foreman to  manage rackhd process.
see GG https://groups.google.com/forum/#!topic/rackhd/UfyZ1v9Gdc4

This PR should be merged at the same time with https://github.com/RackHD/RackHD/pull/479  after new version vagrant box be updated.

Related PR:
* https://github.com/RackHD/RackHD/pull/478  edit packer script, change foreman to pm2.
* https://github.com/RackHD/RackHD/pull/479  edit RackHD readme 
* https://github.com/RackHD/on-build-config/pull/31 edit Vagrantfile for smoke-test
@panpan0000

### CODE REVIEW

**Comment by panpan0000** [comment]
what will happen when user issue multiple ```pm2 start``` ?

**Comment by changev** [comment]
@panpan0000 if use different user , there will be multiple conflicting  process but one user just can start one process for each service.

File: docs/rackhd/ubuntu_source_installation.rst [hunk 1]
```diff
@@ -23,12 +23,14 @@ You may need to update this and /etc/dhcpd.conf to match your local network
 configuration.
 
 This will install all the relevant dependencies and code into ~/src, expecting
-that it will be run with node-foreman.
+that it will be run with `pm2`_.
```
Comment (yyscamper) [inline docs/rackhd/ubuntu_source_installation.rst]: Suggest to add some frequently used command here, so user will feel easier to use it. some example below:
- sudo pm2 restart all: restart all RackHD services
- sudo pm2 restart on-taskgraph: restart the on-taskgraph service only.
- sudo pm2 logs: show the combined real-time log for all RackHD services
- sudo pm2 logs on-taskgraph: show the on-taskgraph real-time log
- sudo pm2 logs flush: clean the RackHD logs
- sudo pm2 status: Show the status of RackHD services

File: docs/rackhd/ubuntu_source_installation.rst [hunk 2]
```diff
@@ -23,12 +23,25 @@ You may need to update this and /etc/dhcpd.conf to match your local network
...
+
+Some useful commands of pm2:
+
+.. code::
+
+    sudo pm2 restart all           # restart all RackHD services
+    sudo pm2 restart on-taskgraph  # restart the on-taskgraph service only.
+    sudo pm2 logs                  # show the combined real-time log for all RackHD services
+    sudo pm2 logs on-taskgraph     # show the on-taskgraph real-time log
+    sudo pm2 logs flush            # clean the RackHD logs
```
Comment (yyscamper) [inline docs/rackhd/ubuntu_source_installation.rst]: It's my fault, this command should be `sudo pm2 flush`

**Pull request merged**